### PR TITLE
pairings

### PR DESCRIPTION
**Pull request by lukepatterson31**

### CODE REVIEW

File: pairings.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import random
+
+
+#assigned pairings
+pairs = []
+
+def pair_random(players): 
+    if len(players)%2 != 0: players.append('None')
```
Comment (piotrberlowski) [inline pairings.py]: Use a  `None` literal as opposed to `'None'` as string.

File: pairings.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import random
+
+
+#assigned pairings
+pairs = []
+
+def pair_random(players):
```
Comment (piotrberlowski) [inline pairings.py:3]: Python will pass lists as references. If you modify the list, your changes will be seen outside of this function. You need to copy the list that was given to you to not create side-effects.

File: pairings.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+def pair_random(players): 
+    if len(players)%2 != 0: players.append('None')
+    random.shuffle(players)
+
+    #create pairing tuples and add to pairings list
+    while len(players) > 1 :  
+        pairs.append((players[0], players [-1]))
+        players.pop(-1)
+        players.pop(0)
```
Comment (piotrberlowski) [inline pairings.py]: First of all, you shouldn't be indexing pop. It just takes an element from the end of a list. You're taking different element than expected here.

Secondly, pop returns the element it took. You can use it to construct the pair.

Thirdly, you can use comprehension to construct the list. Consider:
```suggestion
   pairs = [ (c.pop(), c.pop()) for _ in range(len(c)//2) ]
   if c: pairs.append((c.pop(), None))
```

**Review by piotrberlowski — CHANGES_REQUESTED** [review]
I added some review explanations. Also, as a homework, tell me what this does and how:
```
from itertools import zip_longest

p = [1,2,3,4,5]
args = [iter(p)] * 2
print(list(zip_longest(*args)))
```

File: pairings.py (hunk 3 above)
Comment (lukepatterson31) [reply pairings.py]: I amended the code to this:
`import random

def pair_random(players): 
    
    random.shuffle(players)
   
    pairs = [ (players.pop(), players.pop()) for _ in range(len(players)//2) ]
    if len(players)%2 != 0: pairs.append((players.pop(),None))`

it's returning none, have i done something wrong?

Comment (lukepatterson31) [reply pairings.py]: Fixed it, forgot to return pairs.
It works

**Pull request closed without merging**